### ISSUE TITLE
Partially closed generic type name is not parsed correctly when first type argument is missing

### ISSUE THREAD

**Issue by jhinder** [dotnet/roslyn#71379]
**Version Used**: VS 17.8.2

**Steps to Reproduce**:
```csharp
public class C {
    public void M<T1, T2>() {
        M<, T2>();
        G<, T2> x;
    }
}

public class G<T1, T2>;
```

[SharpLab repro](https://sharplab.io/#v2:EYLgtghglgdgPgAQEwEYCwAoBBmABM3AYVwG9NcL88EAWXAWQB4AVFAGl2aQD4AKASlLlKIphy59+AbmEiKAcUbieuAB4yMIgL6YdGTDnxJci1su5SgA)

**Expected Behavior**:
The type argument list is parsed correctly, with the first type argument being reported as missing (with an appropriate error diagnostic, possibly CS1031 "Type expected"?).

**Actual Behavior**:
The angled brackets cause the expressions to be parsed as greater than/less than expressions, which leads to an unexpected syntax tree with lots of misleading diagnostics.
(The trees of both expressions look quite similar to that from #67277.)

![grafik](https://github.com/dotnet/roslyn/assets/823398/9215910d-d960-4ca9-942b-6b5757a70bd5)
